### ISSUE TITLE
Add disk access programmatically (Sandbox)

### ISSUE THREAD

**Issue by shoopdawoop** [aonez/Keka#222]
Is there a way to do this?

**Comment by gingerbeardman** [comment]
Not that I'm aware of. 

The whole point is these the user has to interact to grant permissions so they're aware of the access the app has.

**Comment by aonez** [comment]
@shoopdawoop as @gingerbeardman said there's no way, the user need to grant access. But it should be done just one time. You have any issue?

**Comment by shoopdawoop** [comment]
I guess if it can't be done, it can't be done, so feel free to close this.

But imagine you are an administrator who wants to deploy Keka to a couple of hundred users via Jamf or Munki, you are going to have a bad time.

**Comment by gingerbeardman** [comment]
>I guess if it can't be done, it can't be done, so feel free to close this. ...

@shoopdawoop in this instance can you not distribute the regular version rather than the MAS version? I assume only the MAS version had the sandbox access request?

**Comment by shoopdawoop** [comment]
No, it was the regular version downloaded from keka.io and it was asking for disk access.

**Comment by gingerbeardman** [comment]
In that case I think it should not. what do you think @aonez ?

**Comment by shoopdawoop** [comment]
In my experience most users will at least be confused by the dialogue, possibly even suspecting malware trying to do something nasty.
Also, iirc, administrator credentials are needed to grant the permission, which many users in managed environments do not have, so this would mean lots of extra work for sysadmins.
If the granting permissions step could be done in an install script, that would be fine imo.

**Comment by aonez** [comment]
> administrator credentials are needed to grant the permission

This is not needed, even for the Guest user.

The other point is true, and my biggest concern. About the script, the entitlements used when signing Keka can be modified and Keka re-signed. Just talking about theory, but one can even disable sandbox.
